src/protego.py

```diff
@@ -97,20 +108,59 @@ def applies_to(self, robotname):
             return len(self.user_agent)
         return 0
 
-    def _quote_path(self, path, safe='/'):
+    def _unquote(self, url, ignore='', errors='replace'):
+        """Replace %xy escapes by their single-character equivalent."""
```

Comment:
Maybe the description of the method should explain that this is a fork of Python’s `urllib.parse.unquote` that includes `ignore` support? (if that’s what this is)

Reply:
This method doesn't have all the functionalities of `urllib.parse.unquote` (such as support for different character encodings since Protego doesn't need it).  Would it be okay to call it a fork ?

I had to write it from scratch with many things borrowed from `urllib.parse.unquote` implementation, since -

1. The Python 3 implementation of `unquote` was not compatible with Python 2. Neither was Python 2 implementation compatible with Python 3.
1. I found the code to be complex and difficult to understand.

Currently the method passes all tests case borrowed from CPython implementation.

Reply:
It was just a though :slightly_smiling_face: